eng/targets/Settings.props

```diff
@@ -50,6 +50,13 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <!-- Nuget Settings -->  
+  <PropertyGroup>
+    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile> <!-- Enable deterministic restore -->
+    <RestoreUseStaticGraphEvaluation>true</RestoreUseStaticGraphEvaluation> <!-- Enable faster restore times -->
+    <RestoreLockedMode Condition="'$(OfficialBuild)' == 'true'">true</RestoreLockedMode>
```

Comment:
Should this condition be `IsContinuousIntegrationBuild`?

Reply:
do we want to turn this on for all CI and not just official builds? I was taking a conservative approach